corehq/ex-submodules/casexml/apps/phone/restore.py

```diff
@@ -124,7 +124,7 @@ def append(self, xml_element):
             self.num_items += num - 1
             self.response_body.write(xml_element)
         else:
-            self.response_body.write(xml_util.tostring(xml_element))
+            self.response_body.write(ElementTree.tostring(xml_element, encoding='utf-8'))
```

Comment:
This part is the only part i am a little uncomfortable with. I only addressed the case and fixture providers (which both returned bytes). Other parts of the restore returned `Element`s and this  does not change them. That means restore providers still return `Element`s (from `ElementTree`) or bytes, but a few of the byte ones use `lxml` behind the scenes. This is not necessarily a clean global switch to lxml.

Reply:
Can you say more about what in particular you're uncomfortable with about it?

I think this is one of the best-tested parts of the codebase so I think if the tests pass it should be fine?

Reply:
I don't love that some parts of the restore use python core `ElementTree` and some now use `lxml`, even if the two don't mix (the lxml portions all convert to string before the code reaches here, at least one `ElementTree` portion does not). I am confident it works, both from my testing and the tests you mentioned, but it seems like a potential area for confusion.